File: coding_freshmen/JAVA/PROFICIENT_TRADER/PROFICIENT_TRADER.java
public class Main {
    public static void main(String[] args) {

        int[] arr = {7,1,5,3,6,4};
        System.out.println(maxProfit(arr));
    }
    static int maxProfit(int[] arr){
        int max = 0, profit = 0;
        for (int i = 0; i < arr.length-1; i++) {
            profit = arr[i+1] - arr[i];
            if (profit > max){
                max = profit;
            }
        }
        return max;
    }
}

File: coding_freshmen/PYTHON/KALINGA_KHATUA/PROFICIENT_TRADER.py

arr = input("Input the array of prices with a space between each price \n(eg. 7 1 5 3 6 4) \n: ")
arr = list(map(int, arr.split(' ')))

min_item = arr[0]
max_profit = 0

for item in arr:
	if min_item > item:
		min_item = item
		
	if max_profit < item - min_item:
		max_profit = item - min_item

print(max_profit)
